tests/typecheck/managers/querysets/test_from_queryset.yml

```diff
@@ -675,3 +675,92 @@
+        # the dynamic class hook for classes with generic arguments..
+        # Revealed type is "myapp.models.ManagerFromQuerySetGeneric[myapp.models.MyModel]"
+        main:15: note: Revealed type is "myapp.models.UnknownManager[myapp.models.MyModel]"
+        myapp/models:50: error: Could not resolve manager type for "myapp.models.MyModel.qs_generic_as_manager"
+    installed_apps:
+        - myapp
+    files:
+        - path: myapp/__init__.py
+        - path: myapp/models.py
+          content: |
+              from typing import TypeVar
+              from django.db.models import Manager, Model, QuerySet
+              from django.db.models.manager import BaseManager
+
+              class QuerySetPopulated(QuerySet["MyModel"]):
+                  ...
+
+              T = TypeVar("T", bound=Model)
+              class QuerySetGeneric(QuerySet[T]):
+                  ...
+
+              class BaseManagerPopulated(BaseManager["MyModel"]):
+                  ...
+
+              class BaseManagerGeneric(BaseManager[T]):
+                  ...
+
+              class ManagerPopulated(Manager["MyModel"]):
+                  ...
+
+              class ManagerGeneric(Manager[T]):
+                  ...
+
+              BaseManagerPopulatedFromQuerySetPopulated = BaseManagerPopulated.from_queryset(QuerySetPopulated)
+              BaseManagerGenericFromQuerySetPopulated = BaseManagerGeneric.from_queryset(QuerySetPopulated)
+
+              BaseManagerPopulatedFromQuerySetGeneric = BaseManagerPopulated.from_queryset(QuerySetGeneric)
+              BaseManagerGenericFromQuerySetGeneric = BaseManagerGeneric.from_queryset(QuerySetGeneric)
+
+              ManagerPopulatedFromQuerySetPopulated = ManagerPopulated.from_queryset(QuerySetPopulated)
+              ManagerGenericFromQuerySetPopulated = ManagerGeneric.from_queryset(QuerySetPopulated)
+
+              ManagerPopulatedFromQuerySetGeneric = ManagerPopulated.from_queryset(QuerySetGeneric)
+              ManagerGenericFromQuerySetGeneric = ManagerGeneric.from_queryset(QuerySetGeneric)
+
+              class MyModel(Model):
+                  bm_populated_from_qs_populated = BaseManagerPopulatedFromQuerySetPopulated()
+                  bm_generic_from_qs_populated = BaseManagerGenericFromQuerySetPopulated()
+
+                  bm_populated_from_qs_generic = BaseManagerPopulatedFromQuerySetGeneric()
+                  bm_generic_from_qs_generic = BaseManagerGenericFromQuerySetGeneric()
+
+                  m_populated_from_qs_populated = ManagerPopulatedFromQuerySetPopulated()
+                  m_generic_from_qs_populated = ManagerGenericFromQuerySetPopulated()
+
+                  m_populated_from_qs_generic = ManagerPopulatedFromQuerySetGeneric()
+                  m_generic_from_qs_generic = ManagerGenericFromQuerySetGeneric()
+
+                  qs_populated_as_manager = QuerySetPopulated.as_manager()
+                  qs_generic_as_manager = QuerySetGeneric["MyModel"].as_manager()
```

Comment:
I think I found a mypy bug here, but I'm not sure. But the `get_dynamic_class_hook` isn't called for `QuerySetGeneric["MyModel"].as_manager()` due to `["MyModel"]`.

If I remove `["MyModel"]` it is triggered. But that means the hook isn't compatible with a classmethod on a generic class.

I have no idea if it's unreasonable to expect it to be triggered, it just feels like a bug that adding type parameters affects the triggering..

Reply:
Please, report it anyways :)

Reply:
(and tag me, since I don't watch all issues)

Reply:
It was already discovered, some time ago. That time from django stubs as well: https://github.com/python/mypy/issues/8359

Reply:
We can add a test-case with `xfail` for now. After `mypy@1.6.0` we will just remove `xfail` and move on.